packages/api/app/index.js

```diff
@@ -13,10 +13,35 @@ const awsServerlessExpressMiddleware = require('aws-serverless-express/middlewar
 
 const { services } = require('@cumulus/aws-client');
 const { MissingRequiredEnvVar } = require('@cumulus/errors');
+const { secretsManager } = require('@cumulus/aws-client/services');
+const Logger = require('@cumulus/logger');
 
 const router = require('./routes');
 const { jsonBodyParser } = require('./middleware');
 
+const log = new Logger({ sender: '@api/index' });
+
+// Load Environment Variables
+// This should be done outside of the handler to minimize Secrets Manager calls.
+const initEnvVarsFunction = async () => {
+  if (!process.env.api_config_secret_id) {
+    log.warn('No process.env.api_config_secret_id set, *not* attempting to provide environment variable defaults');
+    return;
+  }
+  const response = await secretsManager().getSecretValue(
+    { SecretId: process.env.api_config_secret_id }
+  ).promise();
+  if (response.SecretString === undefined) {
+    throw new Error(`AWS Secret did not contain a stored value: ${process.env.api_config_secret_id}`);
+  }
+  const envSecret = JSON.parse(response.SecretString);
+  log.warn('Setting env vars');
+  log.warn(envSecret);
+  process.env = { ...envSecret, ...process.env };
```

Comment:
```suggestion
  try {
    const response = await secretsManager().getSecretValue(
      { SecretId: process.env.api_config_secret_id }
    ).promise();
    if (response.SecretString === undefined) {
      throw new Error(`AWS Secret did not contain a stored value: ${process.env.api_config_secret_id}`);
    }
    const envSecret = JSON.parse(response.SecretString);
    log.warn('Setting env vars');
    log.warn(envSecret);
    process.env = { ...process.env,  ...envSecret };  
  } catch (error) {
    log.error(`Encountered error trying to set environment variables from secret ${process.env.api_config_secret_id}`, error);
    throw error;
  }
```

I'm wondering if this might be preferable because any uncaught exceptions might not get logged very effectively the API gateway handler code. I could be wrong. This makes sure if we get an error from `JSON.parse`, we'll know exactly what happened rather than risk it getting swallowed. But it may be unnecessary

Reply:
I'm not sure it's strictly required, but I'll gladly err on the side of more explicitly logging things: 

89d9e00af